autopilot/scanner.go

```diff
@@ -280,3 +294,11 @@ func (s *scanner) currentTimeout() time.Duration {
 	defer s.mu.Unlock()
 	return s.timeout
 }
+
+func minRecentScanFailures(scanInterval, maxDowntime time.Duration) uint64 {
+	min := defaultMinRecentScanFailures
+	for time.Duration(min)*scanInterval > maxDowntime {
+		min /= 2
```

Comment:
I wouldn't do this. Instead I'd keep the min failures and max downtime completely separate and make sure the min failures are always respected even if the max downtime is smaller than a scan interval.